src/auth.js

```diff
@@ -108,18 +108,24 @@ async function getClientToken(client, method, path, payload) {
 
     core.debug(`Retrieving Vault Token from v1/auth/${path}/login endpoint`);
 
-    /** @type {import('got').Response<VaultLoginResponse>} */
-    const response = await client.post(`v1/auth/${path}/login`, options);
-    if (response && response.body && response.body.auth && response.body.auth.client_token) {
-        core.debug('✔ Vault Token successfully retrieved');
-
-        core.startGroup('Token Info');
-        core.debug(`Operating under policies: ${JSON.stringify(response.body.auth.policies)}`);
-        core.debug(`Token Metadata: ${JSON.stringify(response.body.auth.metadata)}`);
-        core.endGroup();
-
-        return response.body.auth.client_token;
-    } else {
+    
+    try {
+        /** @type {import('got').Response<VaultLoginResponse>} */
+        const response = await client.post(`v1/auth/${path}/login`, options);
+        if (response && response.body && response.body.auth && response.body.auth.client_token) {
+            core.debug('✔ Vault Token successfully retrieved');
+    
+            core.startGroup('Token Info');
+            core.debug(`Operating under policies: ${JSON.stringify(response.body.auth.policies)}`);
+            core.debug(`Token Metadata: ${JSON.stringify(response.body.auth.metadata)}`);
+            core.endGroup();
+    
+            return response.body.auth.client_token;
+        } else {
+            throw Error(`Unable to retrieve token from ${method}'s login endpoint.`);
+        }
+    } catch (error) {
+        core.debug(error.response.body);
```

Comment:
I assume the Hashicorp folks will have more context on the threat model associated with error responses from Vault, but it would be great if we could write this using `core.log` instead of `core.debug` so that we don't need to opt-into debug logging for all actions just to triage an issue.

(For context, I spent some time yesterday tracking down a clock-skew issue which appeared intermittently)